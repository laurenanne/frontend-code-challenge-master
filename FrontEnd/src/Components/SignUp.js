import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import { Input, Button, Col, Form, FormGroup, Label } from "reactstrap";

function SignUp({ signup }) {
  const initialState = {
    email: "",
    firstName: "",
    lastName: "",
  };

  const navigate = useNavigate();
  const [formData, setFormData] = useState(initialState);
  const [formErrors, setFormErrors] = useState([]);

  // handles the submission of the data from form
  async function handleSubmit(evt) {
    evt.preventDefault();

    // calls signup function and if success navigates to admin page
    let result = await signup(formData);
    if (result.success) {
      setFormData(initialState);
      navigate("/admin", { replace: true });
    } else setFormErrors(result.err);
  }

  // handles the change when user input is entered
  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setFormData((data) => ({
      ...data,
      [name]: value,
    }));
  };

  return (
    <div className="m-3 d-flex justify-content-center">
      <Col md={6}>
        <h3 className="text-center">Sign Up</h3>
        <Form onSubmit={handleSubmit}>
          <FormGroup>
            <Label for="email">Email</Label>
            <Input
              id="email"
              type="email"
              name="email"
              onChange={handleChange}
              value={formData.email}
            />
          </FormGroup>

          <FormGroup>
            <Label for="firstName">First Name</Label>
            <Input
              id="firstName"
              type="text"
              name="firstName"
              onChange={handleChange}
              value={formData.firstName}
            />
          </FormGroup>

          <FormGroup>
            <Label for="lastName">Last Name</Label>
            <Input
              id="lastName"
              type="text"
              name="lastName"
              onChange={handleChange}
              value={formData.lastName}
            />
          </FormGroup>
          {formErrors.length > 0 ? (
            <div className="p-2 alert alert-danger">
              {formErrors.map((e) => (
                <p className="mb-0">{e}</p>
              ))}
            </div>
          ) : (
            ""
          )}

          <Button style={{ color: "white" }} className="=btn btn-secondary">
            Submit
          </Button>
        </Form>
      </Col>
    </div>
  );
}

export default SignUp;
