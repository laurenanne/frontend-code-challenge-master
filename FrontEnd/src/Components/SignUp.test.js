import React from "react";
import { render } from "@testing-library/react";
import SignUp from "./SignUp";
import { MemoryRouter } from "react-router";

it("matches snapshot", function() {
  const { asFragment } = render(
    <MemoryRouter>
      <SignUp />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
