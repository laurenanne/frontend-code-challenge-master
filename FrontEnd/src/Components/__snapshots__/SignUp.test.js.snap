// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="m-3 d-flex justify-content-center"
  >
    <div
      class="col-md-6"
    >
      <h3
        class="text-center"
      >
        Sign Up
      </h3>
      <form
        class=""
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="email"
          >
            Email
          </label>
          <input
            class="form-control"
            id="email"
            name="email"
            type="email"
            value=""
          />
        </div>
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="firstName"
          >
            First Name
          </label>
          <input
            class="form-control"
            id="firstName"
            name="firstName"
            type="text"
            value=""
          />
        </div>
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="lastName"
          >
            Last Name
          </label>
          <input
            class="form-control"
            id="lastName"
            name="lastName"
            type="text"
            value=""
          />
        </div>
        <button
          class="=btn btn-secondary btn btn-secondary"
          style="color: white;"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
